src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,583 @@
+                                          pinned_coords, coursier_workdir)
+
+      cmd_str = ' '.join(cmd_args)
+      logger.debug(cmd_str)
+
+      try:
+        with self.context.new_workunit(name='coursier', labels=[WorkUnitLabel.TOOL]) as workunit:
+
+          return_code = self.runjava(
+            classpath=[coursier_jar],
+            main='coursier.cli.Coursier',
+            args=cmd_args,
+            jvm_options=self.get_options().jvm_options,
+            # to let stdout/err through, but don't print tool's label.
+            workunit_labels=[WorkUnitLabel.TOOL, WorkUnitLabel.SUPPRESS_LABEL])
+
+          workunit.set_outcome(WorkUnit.FAILURE if return_code else WorkUnit.SUCCESS)
+
+          if return_code:
+            raise TaskError('The coursier process exited non-zero: {0}'.format(return_code))
+
+          with open(output_fn) as f:
+            result = json.loads(f.read())
+            results[classifier] = result
+
+      except subprocess.CalledProcessError as e:
+        raise CoursierError(e)
+
+    return results
+
+  @staticmethod
+  def _construct_cmd_args(classified_jars, classifier, common_args, global_excludes,
+                          pinned_coords, coursier_workdir):
+    cmd_args = list(common_args)
+
+    # Append classifier if not default
+    if classifier != '':
+      cmd_args.extend(['--classifier', classifier])
+
+    # Dealing with intransitivity and forced versions.
+    for j in classified_jars:
+      if not j.rev:
+        continue
+
+      if j.intransitive:
+        cmd_args.append('--intransitive')
+
+      cmd_args.append(j.coordinate.simple_coord)
+
+      # Force requires specifying the coord again with -V
+      if j.force:
+        cmd_args.append('-V')
+        cmd_args.append(j.coordinate.simple_coord)
+
+    # Force pinned coordinates
+    for j in pinned_coords:
+      cmd_args.append('-V')
+      cmd_args.append(j.simple_coord)
+
+    # Local exclusions
+    local_exclude_args = []
+    for jar in classified_jars:
+      for ex in jar.excludes:
+        # `--` means exclude. See --local-exclude-file in `coursier fetch --help`
+        # If ex.name does not exist, that means the whole org needs to be excluded.
+        ex_arg = "{}:{}--{}:{}".format(jar.org, jar.name, ex.org, ex.name or '*')
+        local_exclude_args.append(ex_arg)
+
+    if local_exclude_args:
+      with temporary_file(coursier_workdir, cleanup=False) as f:
+        exclude_file = f.name
+        with open(exclude_file, 'w') as ex_f:
+          ex_f.write('\n'.join(local_exclude_args).encode('utf8'))
+
+        cmd_args.append('--local-exclude-file')
+        cmd_args.append(exclude_file)
+
+    for ex in global_excludes:
+      cmd_args.append('-E')
+      cmd_args.append('{}:{}'.format(ex.org, ex.name or '*'))
+
+    return cmd_args
+
+  def _load_json_result(self, classifier, compile_classpath, coursier_cache_path, invalidation_check,
```

Comment:
This deserves a method doc, particularly since it has an argument named "result".